src/mscorlib/src/System/Diagnostics/Eventing/XplatEventLogger.cs

```diff
@@ -79,6 +79,10 @@ private static string Serialize(ReadOnlyCollection<string> payloadName, ReadOnly
                 return String.Empty;
 
             Contract.Assert(payloadName.Count == payload.Count);
+            if(payloadName.Count != payload.Count)
```

Comment:
Yes, this will fix things up for release builds.  My goal for this change is to unblock tracing on release builds while an investigation is done into the root cause.  I've filed #2233 to track root causing this.